docs/weave/tutorial/11-cron.md

```diff
@@ -0,0 +1,325 @@
+			return nil, errors.Wrap(err, "cannot store delete article task")
+		}
+	}
+```
+
+If the create message contains a DeleteAt value, `deleteArticleMsg` will be initiated and
+scheduled with no condition. Don't get scared by **with no condition** words, the CRON router
+will not be accessible to the world anyways.
+
+Let's say we want a message that could be executed only when an **admin's** condition is present
+in the context. For this, the handler that will execute message must check the if the admin's
+condition is in the context and then execute accordingly. In order to pass this authentication
+information to the CRON handler you need to feed the condition to the scheduler:
+
+`taskID, err = h.scheduler.Schedule(store, article.DeleteAt.Time(), adminCond, deleteArticleMsg)`
+
+In the code example above we also defined a `taskID` that will be used to identify the task among
+other deletion tasks, with this information we can delete or modify the task after scheduling
+process is over.
+
+As you can see `deleteArticleTask` is saved to the store with article ID and task owner address
+to authorize modification to the task.
+
+Via the article ID field, scheduled tasks could be queried with articles.
+
+## Executor Handler
+
+Executor Handler in this context refers to the handler that will execute the task that is
+received from the application ticker. We implemented the deletion scheduling logic within
+`CreateArticleHandler`, now we have to implement the consumer logic, which is
+`CronDeleteArticleHandler`.
+
+`Deliver` method of `CronDeleteArticleHandler`:
+
+```go
+// Deliver stages a scheduled deletion if all preconditions are met
+func (h CronDeleteArticleHandler) Deliver(ctx weave.Context, store weave.KVStore, tx weave.Tx) (*weave.DeliverResult, error) {
+	msg, err := h.validate(ctx, store, tx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := h.b.Delete(store, msg.ArticleID); err != nil {
+		return nil, errors.Wrapf(err, "cannot delete article with ID %s", msg.ArticleID)
+	}
+
+	return &weave.DeliverResult{}, nil
+}
+```
+
+As you can see there is no authentication checks, it is because we designed cron handlers to be
+inaccesible to the world via a query so we can implement our execution without any fears of
+exploitation.
+
+This handler will receive the `DeleteArticleMsg`, delete the article from the store using the
+article ID data that is passed along with the message.
+
+## Task Cancellation
+
+If we need the ability to do execution in distant future, we also need the ability to cancel the
+same execution. As you remember we saved the scheduled task information to the store to be used
+in various post-submission activities. Deletion is the simplest example of this case. As you read
+many many times, and probably got very used to, every action in Weave goes through an handler. We
+will create a handler that is called `CancelDeleteArticleTaskHandler`.
+
+`validate` method:
```

Comment:
This line is not needed, as the first line of code starts with the method name :wink: